src/utils/recommendationUtils.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright 2023 Red Hat, Inc. and others.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  IRecommendationService,
+  RecommendationCore,
+} from "@redhat-developer/vscode-extension-proposals";
+import { ExtensionContext } from "vscode";
+
+const OPENSHIFT_TOOLKIT_ID = "redhat.vscode-openshift-connector";
+
+let recommendationService: IRecommendationService;
+
+/**
+ * Initialize the recommendation service
+ *
+ * @param context the extension context
+ */
+export function initRecommendationService(context: ExtensionContext) {
+  if (!recommendationService) {
+    recommendationService = RecommendationCore.getService(context);
```

Comment:
If you look at the API, you'll see that `getService` takes an optional telemetry service. If you provide the telemetry service that was initialize just above where this method is called, the results of the recommendation will be recorded.